### PR TITLE
add mathjs

### PR DESCRIPTION
**Pull request by slavaGanzin**
[Mathjs](https://github.com/josdejong/mathjs) is kinda calculator with some neat features
2019-07-08-164028_955x239_scrot

Use case is smth like: https://soulver.app/

### CODE REVIEW

File: autoload/codi/load.vim [hunk 1]
```diff
@@ -193,6 +193,10 @@ let s:codi_default_interpreters = {
          \ 'bin': ['iex'],
          \ 'prompt': '^iex(\d\+). '
          \ },
+      \ 'math': {
```
Comment (metakirby5) [inline autoload/codi/load.vim]: Can we change this to `mathjs` to disambiguate?

**Comment by metakirby5** [comment]
Thanks for your contribution 😄 
I pushed some whitespace fixes to your branch and left some comments to address.

**Comment by metakirby5** [comment]
> I noticed that the expression doesn't actually evaluate until we feed another newline to the buffer. Can you look into why that is?

Actually, it seems that the issue only persists until a second line is edited - if the second line is deleted, the first line will continue to evaluate until exit. There might be some other underlying issue, but it's low priority and shouldn't block this PR. Will merge once we resolve the comment about `math -> mathjs`.

**Comment by slavaGanzin** [comment]
@metakirby5 I, by error, rewrite your commits, but it seems it looks the same.

**Comment by metakirby5** [comment]
Thanks for the changes! Final thing: can you add documentation to the following places?

- `README.md`: Line after "Languages with built-in support:"
- `doc/codi.txt`: Table in codi-interpreters-defaults
  - For this one, perhaps with the note that evaluation may not work until a newline is entered.

**Comment by slavaGanzin** [comment]
@metakirby5 Ethan, anything else needed to merge this?

**Comment by metakirby5** [comment]
> @metakirby5 Ethan, anything else needed to merge this?

Yes - can you make the changes I requested above in `README.md` and `doc/codi.txt`?

**Comment by slavaGanzin** [comment]
@metakirby5 Is it ok? https://github.com/metakirby5/codi.vim/pull/112/commits/fdb54c6023347111a1795762bf4b00257d50c606

**Comment by metakirby5** [comment]
Oops, I guess I missed those changes for some reason. Looks good, thanks!

**Pull request merged**